src/compiler/checker.ts

```diff
@@ -29350,13 +29351,19 @@ namespace ts {
                 (<PrefixUnaryExpression>expr).operand.kind === SyntaxKind.NumericLiteral;
         }
 
+        function isSimpleLiteralEnumReference(expr: Expression) {
+            if (!(isPropertyAccessExpression(expr) || (isElementAccessExpression(expr) && isStringOrNumberLiteralExpression(expr.argumentExpression)))) return false;
+            if (!isEntityNameExpression(expr.expression)) return false;
+            return !!(checkExpressionCached(expr).flags & TypeFlags.EnumLiteral);

```

Comment:
Why three return statements when one will do?

Reply:
Eh. Sometimes my preference is to break a long condition up using statements instead of ternaries/and+or just to make it a bit easier to read/edit (since it gets optimized to the same thing anyway, this just lets you place breakpoints between more things when debugging it). In fact, it's my normal - usually I only go back in and compact a set of statements once I'm confident I no longer need to adjust them. I can combine them, if you'd prefer.